### PR TITLE
EDGOAIPMH-101: Update to Java 17.

### PR DESCRIPTION
**Pull request by DmytroBykov1**
## Purpose
 
Update the module to Java 17 and identify issues.
## Approach
 
Module updated to Java 17.
## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [ ] Code coverage on new code is 80% or greater
   - [x] Duplications on new code is 3% or less
   - [ ] Check logging
   - [x] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [x] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,9 +29,8 @@
   </licenses>
 
   <properties>
-    <raml-module-builder.version>35.0.1</raml-module-builder.version>
-    <maven.compiler.source>11</maven.compiler.source>
-    <maven.compiler.target>11</maven.compiler.target>
+    <java.version>17</java.version>
+    <raml-module-builder.version>35.0.6</raml-module-builder.version>
```
Comment (khandramai) [inline pom.xml:33]: Do we still need RMB dependency in pom.xml?

File: ramls/traits/resps.raml [hunk 2]
```diff
@@ -3,22 +3,24 @@
         description: 'Success'
         body:
           text/xml:
-            schema: OAI-PMH
+            #schema: OAI-PMH
```
Comment (khandramai) [inline ramls/traits/resps.raml:6]: This can be removed instead of using comment here and below.

File: ramls/edge-oai-pmh.raml [hunk 3]
```diff
@@ -7,7 +7,7 @@ documentation:
     title: 'Edge API - OAI-PMH'
     content: 'Edge API to interface with FOLIO for 3rd party harvesters to harvest metadta via OAI-PMH'
 types:
-  OAI-PMH: !include schemas/OAI-PMH.xsd
+#  OAI-PMH: !include schemas/OAI-PMH.xsd
```
Comment (khandramai) [inline ramls/edge-oai-pmh.raml:10]: I can propose to remove this instead of using comment.

File: pom.xml (hunk 1 above)
Comment (DmytroBykov1) [reply pom.xml:33]: Yes, we do need it. It processes ramls/edge-oai-pmh.raml - cfg and produces the following dependency set 
image

File: ramls/edge-oai-pmh.raml (hunk 3 above)
Comment (DmytroBykov1) [reply ramls/edge-oai-pmh.raml:10]: done

File: pom.xml [hunk 4]
```diff
@@ -196,6 +194,13 @@
         <groupId>org.jvnet.jaxb2.maven2</groupId>
         <artifactId>maven-jaxb2-plugin</artifactId>
         <version>${maven.jaxb2.plugin.ver}</version>
+        <dependencies>
+          <dependency>
+            <groupId>org.glassfish.jaxb</groupId>
+            <artifactId>jaxb-runtime</artifactId>
+            <version>2.3.5</version>
```
Comment (alekGbuz) [inline pom.xml:201]: @DmytroBykov1,  please move dependency version to properties.

**Review by alekGbuz — APPROVED** [review]

File: pom.xml (hunk 4 above)
Comment (DmytroBykov1) [reply pom.xml]: done

**Review by obozhko-folio — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**